src/tevatron/trainer.py

```diff
@@ -20,9 +20,9 @@
     _grad_cache_available = False
 
 
-class DenseTrainer(Trainer):
+class EncoderTrainer(Trainer):
```

Comment:
Now I think about it, EncoderTrainer still feels a little bit weird. Can we just call this Trainer? If not, maybe Tevatron trainer?